### ISSUE TITLE
[Keyboard Navigation-Project -Makecode Documentation] Focus is not naviagating to first interactive element after selecting the any link in the left pane.

### ISSUE THREAD

**Issue by v-mesala** [Microsoft/pxt#2807]
**User Experience:**
User who depends on Keyboard with screen reader may get misguided if the Focus is not navigating to first interactive element after selecting the any link in the left pane..

**Test Environment:**
OS: RS2 Version 1703(OS Build 15063.2)
Platform: Edge.

**Repro Steps:-**
1. Navigate to https://makecode.microbit.org/projects
2. Select any link in the left pane and verify the issue.

**Actual Result:-**
Focus is not navigating to first interactive element after selecting the any link in the left pane.

**Expected Result:-**
Focus should navigate to first interactive element after selecting the any link in the left pane.
If we select Flashing heart link then focus should navigate to first interactive element like Flashing Heart  link after Projects.

**Note:** 
This issue exists with Firefox also.

**MAS Reference -**
https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={0de7fbe1-ad7e-48e5-bcbb-8d986691e2b9}

**Suggested Fix:**
In all the applicable forms we need to check below-  
1.Check if 'tabindex' is used 
2.If 'tabindex' is used, check that the tab order specified by the 'tabindex' attributes follows relationships in the content. 
 
tabindex property is applicable to all visual elements on screen.  
<a href="/articles/" tabindex="2">Articles</a> 
<a href="/resources/" tabindex="3">Resources</a> 
 
Tips : 
'tabindex' should only be used in cases where: 
•The default tab order is not ideal, AND 
•The tab order cannot be changed by rearranging items in the content and/or by altering the style sheet to reflect the best visual arrangement.
Additional Note: Consider whether content may be changed, added, or removed frequently on the web page. If so and if content order changes from time to time, then using tabindex="0" may also be an option in order to minimize maintenance. Content that has tabindex="0" set on it will cause the tabbing order to the follow the order of the of the content in the source code.
The links in the snippet below all use the tabindex="0" attribute, which provides a logical tabbing order that matches the visual order and source code order or the links.

**Refer Attachment:**
[MAS 2.4.3_MakeCode_Focus is not navigating to first interactive element.zip](https://github.com/Microsoft/pxt/files/1248891/MAS.2.4.3_MakeCode_Focus.is.not.navigating.to.first.interactive.element.zip)

**Comment by veler** [comment]
@microsoftsam , according to this issue, for me, that's why we have the hidden menu. What do you think about? Fixed or won't fix?

**Comment by veler** [comment]
After discussion with @microsoftsam , we are agree that we already fixed this issue thanks to a hidden menu. But it is not deployed yet.

**Comment by v-mesala** [comment]
Verified on URL:https://makecode.microbit.org/projects and issue is not fixed now. Hence reactivating the bug.

**Comment by veler** [comment]
We decided that the hidden menu was replacing an automatic focus to the content. @microsoftsam and @abchatra , should we make an exception for this issue?

**Comment by samelhusseini** [comment]
Hi @v-mesala I've looked into this issue myself and I disagree that this is an issue. The sidebar menu is a menu with links to different pages, when a user navigates to a certain page there's no way for us to know that this is a user that is navigating through the sidebar vs a user that is just navigating to the page and hence can't just focus them directly onto the content as they may want to navigate to a different page through the side menu.

In order to make it easier to navigate to the content, we've added a hidden menu that when a user presses tab, they're able to "Skip to main content". 

Please look into this and let me know your thoughts.

**Comment by samelhusseini** [comment]
@v-amik could you please also look into this. I'd like to resolve this issue.

**Comment by abchatra** [comment]
As per our understanding and above description from Sam, this is by design and current solution is better for accessibility story. We can discuss this if you have concerns. Closing for now.
